src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -514,15 +569,12 @@ public static Vector128<ushort> Load128(ref byte ptr)
                 {
                     return AdvSimd.ZeroExtendWideningLower(Vector64.LoadUnsafe(ref ptr));
                 }
-                else if (Sse2.IsSupported)
-                {
-                    Vector128<byte> vec = Vector128.CreateScalarUnsafe(Unsafe.ReadUnaligned<long>(ref ptr)).AsByte();
-                    return Sse2.UnpackLow(vec, Vector128<byte>.Zero).AsUInt16();
-                }
                 else
                 {
-                    (Vector64<ushort> lower, Vector64<ushort> upper) = Vector64.Widen(Vector64.LoadUnsafe(ref ptr));
-                    return Vector128.Create(lower, upper);
+                    Debug.Assert(Vector128.IsHardwareAccelerated);
```

Comment:
I'm not against that assert, but do we need it?

Reply:
Not really, I will remove it